src/App/Repository/User/SettingRepository.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+/**
+ * Ushahidi User Setting Repository
+ *
+ * @author     Ushahidi Team <team@ushahidi.com>
+ * @package    Ushahidi\Application
+ * @copyright  2018 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+namespace Ushahidi\App\Repository\User;
+
+use Ohanzee\DB;
+use Ushahidi\Core\Data;
+use Ushahidi\Core\Entity;
+use Ushahidi\Core\SearchData;
+use Ushahidi\Core\Entity\UserSetting;
+use Ushahidi\Core\Entity\UserSettingRepository as UserSettingRepositoryContract;
+use Ushahidi\App\Repository\OhanzeeRepository;
+use Ushahidi\App\Repository\JsonTranscodeRepository;
+
+class SettingRepository extends OhanzeeRepository implements
+	UserSettingRepositoryContract
+{
+	// OhanzeeRepository
+	protected function getTable()
+	{
+		return 'user_settings';
+	}
+
+	// CreateRepository
+		// ReadRepository
+	public function getEntity(array $data = null)
+	{
+		return new UserSetting($data);
+	}
+
+	// SearchRepository
+	public function getSearchFields()
+	{
+		return ['user_id'];
+	}
+
+	// OhanzeeRepository
+	protected function setSearchConditions(SearchData $search)
+	{
+		$query = $this->search_query;
+
+		if ($search->user_id) {
+			$query->where('user_id', '=', $search->user_id);
+		}
+	}
+
+	// UserSettingRepository
+	public function updateCollection(array $entities)
```

Comment:
Is this used anywhere yet? I really like this idea ie. that you can overwrite all user settings at once, since that seems likely how it'll get used in the UI anyway

Reply:
No it's only a placeholder so far.